service/history/conflictResolverV2.go

```diff
@@ -0,0 +1,186 @@
+// Copyright (c) 2017 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package history
+
+import (
+	"fmt"
+	"github.com/uber/cadence/.gen/go/shared"
+	"github.com/uber/cadence/common"
+	"github.com/uber/cadence/common/cluster"
+	"github.com/uber/cadence/common/collection"
+	"github.com/uber/cadence/common/log"
+	"github.com/uber/cadence/common/log/tag"
+	"github.com/uber/cadence/common/persistence"
+)
+
+const (
+	defaultPageSize   = 100
+	eventStoreVersion = persistence.EventStoreVersionV2
+)
+
+type (
+	conflictResolverV2 interface {
+		reset(prevRunID string, requestID string, replayEventID int64, info *persistence.WorkflowExecutionInfo) (mutableState, error)
+	}
+
+	conflictResolverV2Impl struct {
+		shard           ShardContext
+		clusterMetadata cluster.Metadata
+		context         workflowExecutionContext
+		historyMgr      persistence.HistoryManager
+		historyV2Mgr    persistence.HistoryV2Manager
+		logger          log.Logger
+	}
+)
+
+func newConflictResolverV2(shard ShardContext, context workflowExecutionContext, historyMgr persistence.HistoryManager, historyV2Mgr persistence.HistoryV2Manager,
+	logger log.Logger) *conflictResolverImpl {
+
+	return &conflictResolverImpl{
+		shard:           shard,
+		clusterMetadata: shard.GetService().GetClusterMetadata(),
+		context:         context,
+		historyMgr:      historyMgr,
+		historyV2Mgr:    historyV2Mgr,
+		logger:          logger,
+	}
+}
+
+func (r *conflictResolverV2Impl) reset(prevRunID string, requestID string, replayEventID int64, info *persistence.WorkflowExecutionInfo, localVersionHistories persistence.VersionHistories) (mutableState, error) {
+	domainID := r.context.getDomainID()
+	execution := *r.context.getExecution()
+	branchToken := info.GetCurrentBranch()
```

Comment:
this should come from LAC's corresponding branch, for now make it a empty []byte and add a TODO